src/v/cluster/topic_table.h

```diff
@@ -167,8 +167,7 @@ class topic_table {
     using delta_cb_t
       = ss::noncopyable_function<void(const std::vector<delta>&)>;
 
-    explicit topic_table()
-      : _probe(*this){};
+    explicit topic_table(ss::sharded<partition_manager>&);
```

Comment:
topic_table should not depend on `partition_manager` the table is a metadata only type it keeps the topic state, it should not interact with the manager which is responsible for doing the actual partition related work